tools/arch-visualizer/viewer/src/components/MarkdownRenderer.tsx

```
@@ -0,0 +1,187 @@
+ * Handles: headings, bold, italic, code blocks, inline code, links, lists, blockquotes, HR, tables.
+ */
+export function MarkdownRenderer({ content, darkMode }: { content: string; darkMode: boolean }) {
+  const html = useMemo(() => renderMarkdown(content), [content]);
+
+  return (
+    <div
+      className={`
+        prose prose-sm max-w-none
+        ${darkMode ? "markdown-dark" : "markdown-light"}
+      `}
+      dangerouslySetInnerHTML={{ __html: html }}
```

Biome (2.3.13)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)